### ISSUE TITLE
Build a set of example hues

### ISSUE THREAD

**Issue by KirstieJane** [mentalhues/mentalhues.github.io#18]
I think it makes sense to start to build a collection of mental hues to help newcomers to the project understand the idea behind the project.

* Where should these example hues go?
  * As a suggestion, what about in a file in the repository called `example-hues.md`?

Here's one from me to get us started:

### Telling my boyfriend why I had that doctors appointment

I'd been dating my (now) boyfriend for only a few weeks when I had to get up early to go to a doctors appointment. I was pretty nervous about what I was going to say when he asked what was wrong with me: I didn't have a cold and hadn't been complaining about feeling unwell. As it was, he just said goodbye, gave me a kiss and didn't ask any questions. Although that meant I didn't have to answer those questions, I realised that I had actually *wanted* him to ask me what was going on. So that evening, over dinner, I asked him if he wanted to know why I'd gone to the doctors. He said something along the lines of it not being any of his business, which actually made me kinda sad: I wanted it to be his business. I told him that and he said he was happy to know why I went, he just didn't want to be pushy or make me feel uncomfortable (which is exactly what I'd been worried about that morning!)

Long story short, I told him that I'd gone to the docs to get a refill on my anti-depressants, and that although I was doing really well, I felt that it was important for me to still take them as I was still struggling with some symptoms of my depression. (I often sleep too much when I'm depressed and feel like I'm a waste of space. The anti-depressants are really useful to me to break out of the negative thought spirals and actually get on with my day.) He was *so* supportive. He told me about some of his mental health challenges, and how he could understand why I was nervous about telling him. We both felt that that conversation brought us closer and opened up more things we could share with each other.

**Comment by erindcole** [comment]
@KirstieJane this is wonderful. I'd love to include it, and I like the idea of making a folder of sample stories. To answer your questions above, I think a title should be your choice. It does help give a brief idea of your story, but I don't think a title is required. The length should be up to the user as well, although I imagine that most users would keep their stories to around 300 words. 

To me, metadata means that the stories would be searchable, and that we would collect data about the stories. However, in order to ensure a safe space, I don't think we should collect data about the stories. Maybe the only metadata should be the title of the Hue.

Seems to me that this folder should just be an initial collection, and the title of the folder could be the Hue we create? What do you think?

**Comment by KirstieJane** [comment]
@erindcole awesome! I’m glad it’s close to what you were imagining.

I totally understand the point about not breaking the anonymity of the writers by adding too much detail.

I’m not sure I get the link to making them searchable though? Maybe you can explain that a little more?

What I was thinking for metadata would be ways in which we can link up the hues. I’m imagining that once there are more than 50 it won’t be possible to read them all, so the visitor to the site might want to read stories related to each other (and presumably the mental health themes they’re interested in).

So for example, each hue could ask the reader if they’d like to add up to 5 keywords. I would probably add to mine:

* depression
* new relationship
* disclosure
* medication 

Then we could build in functionality to link the hues together to a) recommend next stories and b) show how they overlap.

What do you think?

**Comment by richhx** [comment]
Thanks for sharing the story Kristie! That is a great idea. I think having a directory named `example-hues` would be enough and people can add examples with `*.md`. I see no need to restrict the name of the file if they're just examples for people looking at the source code to understand the purpose of the project. 

With respect to a title, I don't think we should make it optional. A user should be able to easily reference their posts succinctly which is the point of the title IMO. It also provides a brief summary for people to see what they're going to read. Anonymity isnt reallt lost by adding a title. 

Furthermore, I just started reading the issues here, so forgive me if I don't understand the data model yet, but are posts just going to be stored as files on the server? If we want to store data, it should be simple in enough to use a database e.g. MySQL to store the text. Tags/keywords can be filled in by the user, but we should also look into autogenerating these tags based on the text. For example, a simple solution would be to use the most popular words (ignoring filler words). Even a simple `LIKE` filter can work at the given scale we have to search for text in posts. In this manner, we can build a relatively simple web stack that developers can understand and know the purpose/responsibility of the part they're working on (frontend -> web server -> database).

**Comment by erindcole** [comment]
@KirstieJane absolutely let's make a folder and start compiling Hues. I'd be happy to do that, and can make a Hue of my own too. If you like, you can go ahead and make the folder so you can place your already written Hue there.

@richhx I understand your point about the title. That makes sense too. As far as where the Hues are stored, we haven't decided that yet. There are a couple issues about web framework and tools we want to use, as well as info about the static page. You can read them at #12 and #13. If you have suggestions to add, please feel free 😄

**Comment by KirstieJane** [comment]
Hi @erindcole & @richhx!

I've left a couple of reviews on #20. The next task is to update the README and CONTRIBUTING files to clarify what the process for collecting example hues are.

I agree with @richhx's point about a title. I think it will help the reader. Lets keep it in 😸 

I think that once #20 is merged, and I have added my hue as written above, we should create a new issue to further discuss @richhx's point about making it easy to link up and search hues. The point of the `example-hues` folder is to have some test data to build up these ideas - I'm super excited to see where they go! 💯

**Comment by sandan** [comment]
I think PR #23 should expand upon the collection process for example hues. I think there are two ways a user may want to submit an example hue:

- Anonymously. In this case, the user can send their hue to a specific email address. Then their hue will be checked for appropriate content and submitted into the `example-hues/` folder on their behalf by one of the Project Maintainers (PMs). 

- Not Anonymously. In this case, the user can submit a PR against the repo under `example-hues/`. PMs can approve, submit feedback, reject. I think a new label called `hue` can be used for these kinds of PRs since it will to keep track of them all when migrating them to website.

There are a couple of questions that come up:
- Should we use a single e-mail address that users who want anonymity can submit their hues to? Who should have access? A single PM/all PMs?
- When a PM handles an anonymous user's hue, they will have to submit the PR under their name. I think the PR should clearly state that the hue is on behalf of an anonymous user. Maybe something like:
```
On behalf of Anonymous:

> <hue>
```
I think if we refactor the PR template a little, this will make it easier to submit. Is this OK?

**Comment by jeffslofish** [comment]
@sandan For the more static version of the site, for anonymous submissions, there could be a simple form that the submitter fills out that has fields for keywords, the main message body, and anything else that is needed. When submitted, the form would automatically be emailed to one ore more moderators who could then file a pull request on behalf of that person. 

In a more dynamic version of the site it would be cool if they could submit their hue and see it on the site with a notice that it is awaiting moderation, and then when it is approved, everyone could see it.

**Comment by sandan** [comment]
Ok so thinking more about this:

- A PR can be anonymous if the user (not nec. a mod) doesn't mind submitting on behalf of someone else. The PR just has to be attributed properly.
- For right now, collecting example hues (either anonymously or nah) ultimately has to go through the PR process. I think it is important to label these PRs appropriately so we don't lose them when migrating to the real site. Maybe a label called "Hue"?
- @jeffslofish Sounds good to me. I was just thinking of how we could collect hues from the community thru github for now. But we can use [github templates ](https://github.blog/2018-01-25-multiple-issue-and-pull-request-templates/) to give a "proof of concept" (albeit in markdown for the form's content. We could use it as a basis for the actual form in the static site.

[Here is a fork I have with the changes implemented ](https://github.com/mentalhues/mentalhues.github.io/compare/master...sandan:master)(using Erin's email address right now but we can replace it with an official one later). Some things that occurred to me while doing this:

- Do we have submission guidelines for what is appropriate in a Hue? I think the Code of Conduct captures the spirit so I linked it in the PR template for submitting hues (`hue.md`).

Here is a demo Pull Request. Notice the template=hue.md in the URL, this is something that the user has to type unless we make it the default PR template:
- https://github.com/sandan/mentalhues.github.io/compare/master...sandan:example?expand=1&template=hue.md

**Comment by erindcole** [comment]
I appreciate these insights @sandan and @jeffslofish . I completed merging the PR #27 with PR #28 . Not sure of a smoother way to merge a change I made on my forked version of the Mental Hues organization repo. Is there a one step process for merging from a forked branch to an organization branch, and then to a master branch of the organization site?

@sandan , now your fork is behind and needs to be updated to the template I made before. Could you do that and also add your changes? I think at that point it would be great for you to do a PR and merge in your fork, including the edits you made to the other documents. I like the idea of having two PR templates to make a distinction when we collect more hues and stories.
